win32ss/gdi/ntgdi/freetype.c

```diff
@@ -7063,20 +7064,28 @@ IntExtTextOutW(
         bitSize.cx = realglyph->bitmap.width;
         bitSize.cy = realglyph->bitmap.rows;
 
-        /* Do chars other than space and etx have a bitSize.cx of zero? */
-        if (ch0 != L' ' && ch0 != etx && bitSize.cx == 0)
-            DPRINT1("WARNING: WChar 0x%04x has a bitSize.cx of zero\n", ch0);
-
-        /* Don't ignore spaces when computing offset.
-         * This completes the fix of CORE-11787. */
-        if ((pdcattr->flTextAlign & TA_UPDATECP) && ch0 == L' ' && bitSize.cx == 0)
-        { 
-            IntUnLockFreeType();
-            /* Get the width of the space character */
-            TextIntGetTextExtentPoint(dc, TextObj, L" ", 1, 0, NULL, 0, &spaceWidth, 0);
-            IntLockFreeType();
-            bitSize.cx = spaceWidth.cx;
-            realglyph->left = 0;
+        /* Special case for characters that have a bitSize.cx of zero */
+        if (bitSize.cx == 0)
+        {
+            /* Warn about unexpected characters (other than space and ETX) */
+            if (ch0 != L' ')
+            {
+                if (ch0 != ASCII_ETX)
+                    DPRINT1("WARNING: WChar 0x%04x has a bitSize.cx of zero\n", ch0);
+            }
+            /* Account for space character when computing offset.
+             * This completes the fix of CORE-11787. */
+            else if (pdcattr->flTextAlign & TA_UPDATECP)
```

Comment:
```suggestion
            /* Warn about unexpected characters (other than space and ETX) */
            if (ch0 != L' ' && ch0 != ASCII_ETX)
                DPRINT1("WARNING: WChar 0x%04x has a bitSize.cx of zero\n", ch0);

            /* Account for space character when computing offset.
             * This completes the fix of CORE-11787. */
            if (ch0 == L' ' && (pdcattr->flTextAlign & TA_UPDATECP))
```
The compiler should be intelligent enough to do any optimization there.

Reply:
Well. Added as a separate commit, so reviewers can "choose".